core/lib/node/src/task/prometheus_exporter.rs

```diff
@@ -0,0 +1,48 @@
+use prometheus_exporter::PrometheusExporterConfig;
+use zksync_health_check::{CheckHealth, HealthStatus, HealthUpdater, ReactiveHealthCheck};
+
+use super::{IntoZkSyncTask, TaskInitError, ZkSyncTask};
+use crate::node::{NodeContext, StopReceiver};
+
+#[derive(Debug)]
+pub struct PrometheusExporterTask {
+    config: PrometheusExporterConfig,
+    prometheus_health_check: Option<ReactiveHealthCheck>,
+    prometheus_health_updater: HealthUpdater,
+}
+
+impl IntoZkSyncTask for PrometheusExporterTask {
+    type Config = PrometheusExporterConfig;
+
+    fn create(
+        _node: &NodeContext<'_>,
+        config: Self::Config,
+    ) -> Result<Box<dyn ZkSyncTask>, TaskInitError> {
+        let (prometheus_health_check, prometheus_health_updater) =
+            ReactiveHealthCheck::new("prometheus_exporter");
+
+        Ok(Box::new(Self {
+            config,
+            prometheus_health_check: Some(prometheus_health_check),
+            prometheus_health_updater,
+        }))
+    }
+}
+
+#[async_trait::async_trait]
+impl ZkSyncTask for PrometheusExporterTask {
+    fn healthcheck(&mut self) -> Option<Box<dyn CheckHealth>> {
+        self.prometheus_health_check
+            .take()
+            .map(|c| Box::new(c) as Box<dyn CheckHealth>)
+    }
```

Comment:
> Do we have a guarantee that healthcheck will be called no more than once for a task? 

[Yep](https://github.com/matter-labs/zksync-era/blob/776ec26e342993d419ce02ca80b743c82c01f98e/core/lib/node/src/task/mod.rs#L36).

> IMO, defining health check as a part of task initialization (e.g., by providing it into the DI container; see the other comment) could be more straightforward.

Similarly to the comment above, I agree that it'd be nicer, but I would love to find a way where we don't implement anything that is too complex/too generic (since the ROI is not clear), or has user-facing flaws like having to depend on the invocation order (IMHO it's worse). One could argue that [polling a non-fused future ](https://doc.rust-lang.org/std/future/trait.Future.html#panics) also has awkward guarantees, so generally having library-level contracts is OK-ish.